### PR TITLE
Unit tests for health checker action plugin

### PR DESCRIPTION
**Pull request by rhcarvalho**
Raises the code coverage of Python code under `roles/openshift_health_checker`.

### CODE REVIEW

File: roles/openshift_health_checker/test/action_plugin_test.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    assert not failed(result)
+    assert not changed(result)
+    assert not skipped(result)
+
+
+def test_action_plugin_run_check_ok(plugin, task_vars, monkeypatch):
+    check_return_value = {'ok': 'test'}
+    check_class = fake_check(run_return=check_return_value)
+    monkeypatch.setattr(plugin, 'load_known_checks', lambda: {'fake_check': check_class()})
+    monkeypatch.setattr('openshift_health_check.resolve_checks', lambda *args: ['fake_check'])
```
Comment (sosiouxme) [inline roles/openshift_health_checker/test/action_plugin_test.py]: How about extracting these three lines of boilerplate?

Comment (rhcarvalho) [inline roles/openshift_health_checker/test/action_plugin_test.py]: Suggestions for how?

Comment (sosiouxme) [inline roles/openshift_health_checker/test/action_plugin_test.py]: Well, e.g.

```
def patch_checks_with(plugin, monkeypatch, run_return=None, run_exception=None):
    check_class = fake_check(run_return=run_return, run_exception=run_exception)
    monkeypatch.setattr(plugin, 'load_known_checks', lambda: {'fake_check': check_class()})
    monkeypatch.setattr('openshift_health_check.resolve_checks', lambda *args: ['fake_check'])

def test_action_plugin_run_check_ok(plugin, task_vars, monkeypatch):
    check_return_value = {'ok': 'test'}
    patch_checks_with(plugin, monkeypatch, run_return=check_return_value)
```

Passing around the fixtures makes it less pretty I suppose.

Comment (rhcarvalho) [inline roles/openshift_health_checker/test/action_plugin_test.py]: Hmm, that may make it harder to read as you say.

Comment (rhcarvalho) [inline roles/openshift_health_checker/test/action_plugin_test.py]: @sosiouxme, I tried to find a pattern for extracting the 3 lines but indeed it feels like it hurts understanding the test (which I'd say is already non-trivial...). The assertions are also similar in shape but different (true/false), we probably don't benefit much from building a complex table of inputs and expectations without having a color picture to explain what is going on.

Let's go with this and if we ever need to rework the check runner / action plugin we can/will revisit the tests.

**Comment by rhcarvalho** [comment]
aos-ci-test

**Comment by rhcarvalho** [comment]
[merge]

**Comment by rhcarvalho** [comment]
aos-ci-test

**Comment by rhcarvalho** [comment]
`aos-ci-jenkins/OS_unit_tests` marked and failed and no details, I don't understand that.

Screenshot:

image

**Comment by rhcarvalho** [comment]
`aos-ci-jenkins/OS_3.6_containerized — "openshift-ansible install failed"`

Flake or infra problem or both?

**No package matching 'atomic-openshift-docker-excluder-3.6.18\***

```
TASK [openshift_excluder : Install docker excluder] ****************************
fatal: [10.8.170.85]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'atomic-openshift-docker-excluder-3.6.18*' found available, installed or updated", "rc": 126, "results": ["No package matching 'atomic-openshift-docker-excluder-3.6.18*' found available, installed or updated"]}
fatal: [10.8.170.12]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'atomic-openshift-docker-excluder-3.6.18*' found available, installed or updated", "rc": 126, "results": ["No package matching 'atomic-openshift-docker-excluder-3.6.18*' found available, installed or updated"]}
fatal: [10.8.170.88]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'atomic-openshift-docker-excluder-3.6.18*' found available, installed or updated", "rc": 126, "results": ["No package matching 'atomic-openshift-docker-excluder-3.6.18*' found available, installed or updated"]}
	to retry, use: --limit @/home/jenkins/workspace/openshift-ansible-3-test-matrix/CONTAINERIZED/_containerized/OSE_VER/3.6/PYTHON/System-CPython-2.7/TOPOLOGY/openshift-cluster-containerized/TargetBranch/master/nodes/openshift-ansible-slave/openshift-ansible/playbooks/byo/config.retry
```
https://aos-ci.s3.amazonaws.com/openshift/openshift-ansible/jenkins-openshift-ansible-3-test-matrix-CONTAINERIZED=_containerized,OSE_VER=3.6,PYTHON=System-CPython-2.7,TOPOLOGY=openshift-cluster-containerized,TargetBranch=master,nodes=openshift-ansible-slave-1329/cf4a978d230040bed9c2c203bfa0ca0c70d98d64.txt

**Comment by rhcarvalho** [comment]
aos-ci-test

**Comment by rhcarvalho** [comment]
Same problem again...

aos-ci-jenkins/OS_3.6_containerized — "openshift-ansible install failed"

**No package matching 'atomic-openshift-docker-excluder-3.6.18\***

https://aos-ci.s3.amazonaws.com/openshift/openshift-ansible/jenkins-openshift-ansible-3-test-matrix-CONTAINERIZED=_containerized,OSE_VER=3.6,PYTHON=System-CPython-2.7,TOPOLOGY=openshift-cluster-containerized,TargetBranch=master,nodes=openshift-ansible-slave-1330/cf4a978d230040bed9c2c203bfa0ca0c70d98d64.txt

/cc @sdodson, any clue what's going on?

**Comment by sdodson** [comment]
@rhcarvalho Looking into both the unit tests not posting results and the 3.6 jobs. The 3.6 jobs don't block anything currently so merges should still work.

**Comment by sdodson** [comment]
aos-ci-test

**Comment by rhcarvalho** [comment]
Thanks @sdodson! The merge job seems to be still running, for about 4h now (longer than normal?).

https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/169/console

**Comment by sdodson** [comment]
[merge]

**Comment by sdodson** [comment]
Ok, something is off, unit tests were green after the comment was posted but have since gone red.

**Comment by rhcarvalho** [comment]
new day... let's try again...

**Comment by rhcarvalho** [comment]
aos-ci-test

**Comment by rhcarvalho** [comment]
[merge]

**Comment by sosiouxme** [comment]
:tada:

**Pull request merged**